# Домашнє завдання до теми "Основи роботи з рядками та математичні операції"

1. Створіть змінні `a`, `b`, `c` та `d`, задайте їм значення відповідно 101, 210, 111, 402, та обчисліть з допомогою Python наступний вираз:  $$ \frac{(a+b) * c}{d} $$

**Важливо**: тут ми маємо обчислити значення саме з використанням змінних, а не просто чисел, як би ми робили у калькуряторі. Змінні потрібні аби за бажання ми могли змінити їх значення та не переписувати код обчислення виразу, бо він буде працювати для різних значень змінних.

In [1]:
a = 101
b = 210
c = 111
d = 402

In [2]:
result = ((a + b) * c) / d

In [3]:
print(result)

85.8731343283582


Очікувана відповідь: 85.8731343283582

2. Якою буде остача від ділення числа 11283 на 89? Обчисліть з допомогою Python.

In [7]:
e = 11283
f = 89

In [8]:
result_1 = e % f

In [9]:
print(result_1)

69


3. Скільки цілих разів поміститься число 89 в 11283? (Або що те саме: який буде результат цілочисельного ділення 11283 на 89?)

In [10]:
g = 11283
h = 89


In [12]:
result_3 = g/h

In [13]:
print(result_3)

126.7752808988764


4. Запишіть у змінну `name` ваше імʼя.

In [14]:
name = 'Olga'

5. Запишіть у змінну `surname` ваше прізвище.

In [15]:
surname = 'Sumnikova'

6. Сконкатенуйте імʼя та прізвище в одну фразу через пробіл. Запишіть результат у нову змінну `name_surname` та виведіть результат на екран.  

In [18]:
name_surname = name + " " + surname

In [19]:
print(name_surname)

Olga Sumnikova


Очікувана відповідь (тільки з вашими даними): 'Hanna Pylieva'

7. Зробіть усі літери змінної `name_surname` заглавними використовуючи метод рядків в Python:

In [20]:
upper_case_name_surname = name_surname.upper()

Очікувана відповідь (тільки з вашими даними): 'HANNA PYLIEVA'

In [21]:
print(upper_case_name_surname)

OLGA SUMNIKOVA


8. Виведіть третю літеру вашого `name_surname`. Тут уважно, треба саме третя літера. Згадайте, з якого значення починається індексування значень в Python.

In [22]:
name_surname[2]

'g'

Очікувана відповідь (тільки з вашими даними): 'n'

9. Виведіть першу та останню літеру вашого прізвища.

In [23]:
name_surname[0], name_surname[-1]

('O', 'a')

Очікувана відповідь (тільки з вашими даними): ('P', 'a')

10. Виведіть довжину вашого прізвища.

In [24]:
surname_length = len(surname)

Очікувана відповідь (тільки з вашими даними): 7


11. Виведіть кількість літер `a` у столиці Мадагаскару `Antananarivo`. Важливо врахувати всі літери, як великі, так і маленькі.

In [28]:
madagaskar_capital = 'Antananarivo'

In [35]:
madagaskar_capital.lower().count('a')

4

Очікувана відповідь: 4

12. Створіть рядок `my_string` зі значенням: "The quick brown fox jumps over the lazy dog!!!".   
Перевірте, чи починається рядок `my_string` з "The" і чи закінчується він "dog!". Виведіть результати на екран.

In [2]:
my_string = "The quick brown fox jumps over the lazy dog!!!"

In [39]:
start_the_lire = my_string.startswith('The')

In [38]:
my_string.endswith('dog!')

False

In [42]:
print('Does the line starts with "The"?' , {start_the_lire} )
print('Does the line ends with "dog!"?' , {my_string.endswith('dog!')})

Does the line starts with "The"? {True}
Does the line ends with "dog!"? {False}


13. Розділіть рядок `my_string` на список окремих слів, а потім з'єднайте всі слова через нижнє підкреслювання. Виведіть результат на екран.

In [4]:
separate_words = my_string.split()

In [6]:
result_string = "_".join(separate_words)

In [7]:
print(result_string)

The_quick_brown_fox_jumps_over_the_lazy_dog!!!


Очікуваний результат: `The_quick_brown_fox_jumps_over_the_lazy_dog!!!`